pkg/agent/downloader.go

```diff
@@ -29,10 +29,12 @@ import (
 	"path/filepath"
 	"regexp"
 	"strings"
+	"sync"
 )
 
 type Downloader struct {
 	ModelDir  string
+	mu        sync.Mutex
```

Comment:
By the way, if you were to change this to `sync.Mutex` then calling the lock would simply be `d.Lock()`.

Reply:
Thanks, still new to go so never used this feature before.

Reply:
We all start somewhere :)